### PR TITLE
[Bugfix 1863281] fix save password forever not work occasionally issue.

### PR DESCRIPTION
**Pull request by shenqianjin**
[AB#1863281](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1863281)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResource.java [hunk 1]
```diff
@@ -89,7 +89,8 @@ default boolean write(@Nonnull final Element resourceEle, @Nonnull final Databas
             resourceEle.addContent(new Element("url").setText(resource.jdbcUrl.toString()));
             resourceEle.addContent(new Element("username").setText(resource.username));
             resourceEle.addContent(new Element("passwordSave").setText(resource.password.saveType().name()));
-            if (ArrayUtils.isNotEmpty(resource.password.password())) {
+            if (ArrayUtils.isNotEmpty(resource.password.password()) && !StringUtils.equals(String.valueOf(resource.password.password()),
+                    PasswordStore.loadPassword(resource.getType(), resource.getId(), resource.getUsername(), resource.password.saveType()))) {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResource.java:93]: can you extract some variables to make it more readable?
```
final String oldPassword = PasswordStore.loadPassword(resource.getType(), resource.getId(), resource.getUsername(), resource.password.saveType());
final String newPassword = String.valueOf(resource.password.password());
if(StringUtils.isNotEmpty(newPassword) && !ObjectUtils.equals(newPassword, oldPassword)){
  //...
}
```

Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResource.java:93]: what if `newPassword` is empty? will `newPassword` be empty?

Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResource.java]: updated.

Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResource.java]: `newPassword ` is never empty. these is a condition `ArrayUtils.isNotEmpty(resource.password.password())`

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**